### PR TITLE
fix(VirtualizedMessageList): use memoized values as hook dependencies directly

### PR DESCRIPTION
**Pull request by MartinCupela**
### 🎯 Goal

Avoid unnecessary re-renders caused by regenerating a new array of message group style keys (`Object.keys()` was called directly in the hook dependency array).

### CODE REVIEW

**Review by oliverlaz — APPROVED** [review]

**Review by petyosi — CHANGES_REQUESTED** [review]
I believe that we had a bug where `processedMessages` were incorrectly rebuilt with a new object identity at some point, hence the `.length` usage (which, of course, is not correct but we are not aware of any issues because of it). 
Given that, I would say that this change should not touch on that.

**Comment by MartinCupela** [comment]
> I believe that we had a bug where `processedMessages` were incorrectly rebuilt with a new object identity at some point, hence the `.length` usage (which, of course, is not correct but we are not aware of any issues because of it). Given that, I would say that this change should not touch on that.

@petyosi I have reverted the changes and added the explanatory comment for future reference.

**Review by petyosi — APPROVED** [review]

File: src/components/MessageList/VirtualizedMessageList.tsx [hunk 1]
```diff
@@ -206,6 +206,7 @@ const VirtualizedMessageListWithContext = <
         if (style) acc[message.id] = style;
         return acc;
       }, {} as Record<string, GroupStyle>),
```
Comment (arnautov-anton) [inline src/components/MessageList/VirtualizedMessageList.tsx:208]:
```suggestion
      }, {}),
```
Not part of the PR so feel free to ignore this comment but noticed this casting which could be refactored to this:
```ts
array.reduce<Record<string, GroupStyle>>(...)
```

**Review by arnautov-anton — APPROVED** [review]

File: src/components/MessageList/VirtualizedMessageList.tsx (hunk 1 above)
Comment (MartinCupela) [reply src/components/MessageList/VirtualizedMessageList.tsx]: @arnautov-anton added generic type def to all reduce calls

**Review by arnautov-anton — APPROVED** [review]

**Pull request merged**